pkg/models/model_helpers.go

```diff
@@ -22,3 +24,12 @@ func ValueOrEmpty(st *string) string {
 	}
 	return ""
 }
+
+//ConvertEnums converts a pq.StringArray to specific, castable type
+func ConvertEnums[EnumType ~string](pqGroups pq.StringArray) []EnumType {
+	enumValues := []EnumType{}
+	for _, item := range pqGroups {
+		enumValues = append(enumValues, EnumType(item))
+	}
+	return enumValues
+}
```

Comment:
I spent some time playing with this function in the Go playground. I really wanted to find a way to validate that the enum array generated is valid (as auto generated enums do provide a `.IsValid` function that can be used to check if the value is in the allowed list.

I couldn't figure out a way to get this to work from a generics perspective (tried to make the type an Interface, but then had trouble casting), so I just ended up giving up. Not sure if you thought about this at all, but it was an interesting learning experience for me in Go generics and slice types!

For what it's worth, this is probably a moot point because the GQLGen generated resolver code (under the hood) calls `IsValid` before returning the GQL response, and we also have these types strictly bound in our DB, so I can't practically think of a situation where this would actually happen.

Reply:
> auto generated enums do provide a .IsValid function that can be used to check if the value is in the allowed list.

That's interesting, I didn't know of that; are you referring to [this function](https://pkg.go.dev/reflect#Value.IsValid) from the `reflect` package in the stdlib, or something else?

Reply:
@DylanSpOddball https://github.com/CMSgov/mint-app/blob/b8d4482de3b70aab3a1a1e85663be5c7c5028496/pkg/graph/model/models_gen.go#L82-L123

Reply:
Oh, I didn't catch that you were referring to generated code from GQL enums, that makes more sense. Thanks.